frontend_tests/node_tests/stream_data.js

```diff
@@ -475,27 +475,42 @@ run_test('stream_settings', () => {
         history_public_to_subscribers: true,
         stream_post_policy: stream_data.stream_post_policy_values.admins.code,
     };
+
+    const web_public_stream = {
+        stream_id: 4,
+        name: 'd',
+        color: 'red',
+        subscribed: false,
+        invite_only: false,
+        is_web_public: true,
+    };
+
     stream_data.clear_subscriptions();
     stream_data.add_sub(cinnamon);
     stream_data.add_sub(amber);
     stream_data.add_sub(blue);
+    stream_data.add_sub(web_public_stream);
 
     let sub_rows = stream_data.get_streams_for_settings_page();
     assert.equal(sub_rows[0].color, 'blue');
-    assert.equal(sub_rows[1].color, 'amber');
-    assert.equal(sub_rows[2].color, 'cinnamon');
+    assert.equal(sub_rows[1].color, 'red');
```

Comment:
We could potentially make this code more readable by using `assert.equal(sub_rows[1].color, web_public_stream.color);` and similar.

Reply:
Ok. I did a preliminary commit changing the others.